### ISSUE TITLE
syslog or stdout/stderr configurability

### ISSUE THREAD

**Issue by longrunningprocess** [sillsdev/LfMerge#133]
I'd like to see out going to stdout/stderr since we run this in a Docker container and it would be more convenient when troubleshooting to simply look in one place for errors, thanks!

**Comment by marksvc** [comment]
If this has any relation to FieldWorks writing to log files in /tmp/SIL with information that is not also written to stdout/stderr, I would be glad for these logs to (also) be written to stdout/stderr to make it easier to debug, spot problems, and help users supply problem report information. 

If this issue has nothing to do with the above, then never mind :)

**Comment by megahirt** [comment]
@marksvc the logging facility would be specific to LFMerge, which consumes fdo and liblcm.  So no actual impact on FW as far as I understand.

**Comment by rmunn** [comment]
#134 and #135 implement this feature (on the `master` and `fieldworks8-master` branches respectively). Note that currently, the log output will *not* be visible in the container logs, because the LfMerge process runs as user `www-data` and therefore does not have access to the stdout and stderr of PID 1 (which runs as `root`). Once we move the LfMerge process to its own container so it's running as PID 1 in its own container then this shouldn't be as much of an issue. I'm currently looking into a way to make it work, but it looks like we would have to disable some Linux security features to give the LfMerge process write access to PID 1's stdout. So once this is merged, we'll want to have the Language Forge container set the environment variable `LFMERGE_LOGGING_DEST=syslog`, then change the LF entrypoint to run `tail -f /var/log/syslog` in the background so that syslog gets output to stdout.

**Comment by longrunningprocess** [comment]
I was trying to troubleshoot an issue on Language Forge today and had to go to `/var/log/messages` to see this `LfMerge[1866]:    standerr: abort: repository is unrelated` and the over to `/var/lib/languageforge/lexicon/sendreceive/state/` to find `"ErrorMessage": "Putting project 'project_name' on hold due to unhandled exception: \nChorus.VcsDrivers.Mercurial.UnrelatedRepositoryErrorException: The repository(a.k.a 'project' or 'collection') that you tried to synchronize with has the same name as yours, but it does not have the same heritage, so it cannot be synchronized. In order to Send/Receive projects, you have to start with a single project/collection, then copy that around. Don't feel bad if this is confusing, just ask for some technical help.`

Troubleshooting would be much easier if errors were more consolidated into the `_LOGGING_DEST`.  If you all think this would be better captured as a separate issue I can create it but I wanted to see whether this was something that should even be done in the first place.

**Comment by megahirt** [comment]
The primary way I troubleshoot send receive errors is to look for the error in the state file.  You can now use the copy-state script I recently checked in to capture all state files to your local machine :). Then you can open the Json file and look at the message.

Lots of improvements we could make in the process, but that's where things are currently

**Comment by megahirt** [comment]
Related to #201 

@longrunningprocess what would you say the status is on this issue?

**Comment by megahirt** [comment]
> I'd prefer those errors make it into the same place rather than knowing we have to check multiple places to get the whole story. If you agree, we can open another issue for that. Either way I do believe this can be closed.

As far as I understand, the errors you mention and had to find in the state file and in /var/log/messages should all be going into the STDOUT and the container logs now.  So we should be good there.  The state file can be used to see just the actual exception, if you find it difficult to wade through all the container output for the actual error message / exception.

**Comment by megahirt** [comment]
As a side note, I notice the following message in the lfmerge logs on qa.languageforge.org (and presumably production)
`WARNING: syslog logging no longer available, logging to console instead`

Something somewhere in LFMerge is still trying to write to syslog, but it's probably not there anymore (hence the warning)
